p2p/src/net/mod.rs

```diff
@@ -46,9 +49,6 @@ pub trait NetworkingService {
     /// Unique ID assigned to each received request from a peer
     type SyncingPeerRequestId: Debug + Eq + Hash + Send + Sync;
 
-    /// Id that identifies a protocol
-    type ProtocolId: Clone + Debug + Display + Eq + PartialEq + Send;
-
```

Comment:
Out of curiosity, why did you decide to remove this associated type completely instead specifying the newly introduced enum as the `ProtocolId` type in `impl`s?

Reply:
There was a TODO-comment in the `p2p/src/net/types.rs` [file](https://github.com/mintlayer/mintlayer-core/pull/362/files#diff-df7dc133e821c16663d902bb26b75e37f8e19cf19f2d318452ab4aaee4fb224cL52) that said that a protocol id shouldn't be generic. Perhaps I misinterpret it, but I still don't see any benefits in making generic a thing that will probably never be changed.